### PR TITLE
S19

### PR DESCRIPTION
**Pull request by devwistkey**

### CODE REVIEW

File: components/SSI/DeployTydra/index.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            net,
+            'init',
+            'did'
+        )
+        try {
+            const get_free_list = await getSmartContract(
+                init_addr,
+                'tydra_free_list'
+            )
+            const tydra_free_list = await tyron.SmartUtil.default.intoMap(
```
Comment (tralkan) [inline components/SSI/DeployTydra/index.tsx:70]: not sure this map is needed

File: components/SearchBar/index.tsx [hunk 2]
```diff
@@ -354,6 +357,15 @@ function Component() {
                                             }${_username}`
                                         )
                                         break
+                                    case 'initdap':
+                                        Router.push(
+                                            `/${
+                                                _domain === ''
```
Comment (tralkan) [inline components/SearchBar/index.tsx:363]: what id domain = did?

**Review by tralkan — APPROVED** [review]

File: components/SSI/DeployTydra/index.tsx (hunk 1 above)
Comment (tralkan) [reply components/SSI/DeployTydra/index.tsx:70]: map removed, update to match NFT domain free list

File: components/Modals/DashboardModal/index.tsx [hunk 3]
```diff
@@ -489,6 +492,84 @@ function Component() {
...
+        setLoadingDidx(true)
+        const domainId =
+            '0x' + (await tyron.Util.default.HashString(loginInfo?.username))
+        await tyron.SearchBarUtil.default
+            .fetchAddr(net, domainId, 'did')
+            .then(async (addr) => {
+                let res = await getSmartContract(addr, 'version')
+                const version = res.result.version.slice(0, 7)
+                if (
+                    version === 'DIDxWAL' ||
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/Modals/DashboardModal/index.tsx:506]: done

File: components/SSI/DIDxWallet/DIDDocument/Services/index.tsx [hunk 4]
```diff
@@ -226,9 +227,7 @@ function Component() {
                                                                         alt="social-ico"
                                                                     />
                                                                 </div>
-                                                                {element[1][0].split(
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/DIDDocument/Services/index.tsx:229]: Sorry, I've put it back

File: components/SSI/DIDxWallet/Wallet/NFTUsername/DIDDomains/index.tsx [hunk 5]
```diff
@@ -365,6 +365,14 @@ function Component() {
                         navigate(`/${_domain}@${username}`)
                         break
                     default:
+                        updateResolvedInfo({
+                            name: _username,
+                            domain: '',
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/Wallet/NFTUsername/DIDDomains/index.tsx:370]: noted, done

File: components/SSI/DIDxWallet/index.tsx [hunk 6]
```diff
@@ -36,17 +37,28 @@ function Component(props: LayoutProps) {
     const domain = resolvedInfo?.domain
     const isLight = useSelector((state: RootState) => state.modal.isLight)
     const styles = isLight ? stylesLight : stylesDark
+    const is_controller = $isController.getState()
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/index.tsx:40]: done, moved to onclick

File: components/SSI/DIDxWallet/index.tsx [hunk 7]
```diff
@@ -80,7 +99,8 @@ function Component(props: LayoutProps) {
                     <h3 style={{ color: isLight ? '#000' : '#dbe4eb' }}>
                         {docVersion === 'DIDxWAL' ||
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/DIDxWallet/index.tsx:100]: no, but I just add lowercase for all docVersion condition

File: components/Selector/index.tsx [hunk 8]
```diff
@@ -41,11 +59,159 @@ function Selector({
...
+                        }
+                    />
+                </div>
+                <div className={styles.langMobile}>
+                    <Select
+                        menuPlacement={menuPlacement_}
+                        styles={customStylesLangMobile}
+                        theme={(theme) => ({
+                            ...theme,
+                            borderRadius: 0,
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/Selector/index.tsx:177]: added border radius 5px

File: pages/[username]/index.tsx [hunk 9]
```diff
@@ -47,6 +53,26 @@ function Header() {
...
+                                            {domain !== '' &&
+                                                domain !== 'did' &&
+                                                `${domain}@`}
+                                        </span>
+                                        {username}.
+                                        {domain === 'did' ? 'did' : 'ssi'}
+                                    </p>
+                                </h1>
+                            </div>
+                            <Tydra />
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply pages/[username]/index.tsx:75]: done

File: components/SSI/WalletInfo/index.tsx [hunk 10]
```diff
@@ -159,7 +159,10 @@ function Component(props: InputType) {
                             originator_address?.domain
                                 ? originator_address.domain + '@'
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/WalletInfo/index.tsx:160]: Hi, what was the issue? I just test if the domain "" (e.g sibagas) it'll show sibagas.ssi as it should @tralkan

File: components/SearchBar/index.tsx (hunk 2 above)
Comment (devwistkey) [reply components/SearchBar/index.tsx:363]: It'll redirect to did@username

File: components/SSI/WalletInfo/index.tsx [hunk 11]
```diff
@@ -159,7 +159,10 @@ function Component(props: InputType) {
...
                                 : ''}
-                            {originator_address?.username}.did
+                            {originator_address?.username}
+                            {originator_address?.domain === ''
```
[the remark this answers is not in the document]
Comment (devwistkey) [reply components/SSI/WalletInfo/index.tsx:163]: tested, sibagas.did and did@sibagas showed as sibagas.did

**Pull request merged**